src/Hazelcast.Net/CP/AtomicReference.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Threading.Tasks;
+using Hazelcast.Clustering;
+using Hazelcast.Core;
+using Hazelcast.DistributedObjects;
+using Hazelcast.Protocol.Codecs;
+using Hazelcast.Serialization;
+
+namespace Hazelcast.CP
+{
+    internal class AtomicReference<T>: CPDistributedObjectBase, IAtomicReference<T>
```

Comment:
Removed `where T: class` requirement as both Java and JS client support simple primitive types like `int`, `float`, etc.  
The downside is that `await aref.ContainsAsync(await aref.GetAsync());` returns false if invoked with default value, as it's `null` on the server but `default(T)` on the client.

Reply:
That would be consistent with the Java client. It needs to be documented in `GetAsync` and `ContainsAsync` documentation (the `<remarks>` part).

Reply:
Added remarks for `GetAsync` and `GetAndSetAsync`. Not sure if `ContainsAsync` need any changes.